R/OneClassTask.R

```diff
@@ -0,0 +1,30 @@
+#' @export
+#' @rdname Task
+
+makeOneClassTask = function(id = deparse(substitute(data)), data,
+  weights = NULL, blocking = NULL, fixup.data = "warn", check.data = TRUE) {
+  assertString(id)
+  assertDataFrame(data)
+  # some code on cran passed stuff like positive=1, we can live with the convert here
+  # for one class positive is always set to the one class
+  assertChoice(fixup.data, choices = c("no", "quiet", "warn"))
+  assertFlag(check.data)
+
+  task = makeUnsupervisedTask("oneclass", data, weights, blocking,
+    fixup.data = fixup.data, check.data = check.data)
+
+  task$task.desc = makeOneClassTaskDesc(id, data, weights, blocking)
+  addClasses(task, "OneClassTask")
+
+}
+
+makeOneClassTaskDesc = function(id, data, weights, blocking) {
+  target = character(0L)
+  td = makeTaskDescInternal("oneclass", id, data, target, weights, blocking)
+  return(addClasses(td, c("OneClassTaskDesc", "UnsupervisedTaskDesc")))
+}
+
+#' @export
+print.OneClassTask = function(x, ...) {
+  print.UnsupervisedTask(x)
```

Comment:
[`NextMethod()`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/UseMethod.html) automatically jumps to the same function one step up in the class hierarchy. If `x` is a `OneClassTask` with `class(x)` something like `c("OneClassTask", "UnsupervisedTask", "Task")` (I think?) and you evaluate `print(x)`, then R automatically calls `print.OneClassTask` (since that is the first class); if you then call `NextMethod()` within this function (instead of `print.UnsupervisedTask`), it goes to the next step (which would be `print.UnsupervisedTask`, so pretty much what you want). You should try it!

This may not work so well if you call `print.OneClassTask(x)` *directly*, but you probably shouldn't do that anyways.